app/scripts/controllers/detect-tokens.js

```diff
@@ -219,8 +233,9 @@ export default class DetectTokensController {
     this._network = network;
     this._network.store.subscribe(() => {
       if (this.chainId !== this.getChainIdFromNetworkStore(network)) {
-        this.restartTokenDetection();
-        this.chainId = this.getChainIdFromNetworkStore(network);
+        const chainId = this.getChainIdFromNetworkStore(network);
+        this.chainId = chainId;
+        this.restartTokenDetection({ chainId: this.chainId });
```

Comment:
Shouldn't we update the `this.restartTokenDetection()` with params with selectedAddress  in line 69 where it is called when the preference state changes with selectedAddress change as well?

Reply:
yes! I think that change must have gotten lost in a rebase!

Reply:
Great catch!

Reply:
Done [here](https://github.com/MetaMask/metamask-extension/pull/16866/commits/02b67c3cf8b6a357ec81213bf162b5585e716ca1)